File: Leetcode/3375. Minimum Operations to Make Array Values Equal to K.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        unordered_map<int,int> f;
        int mini=INT_MAX;
        for(int i:nums){
            f[i]++;
            mini=min(mini,i);
        }
        cout<<mini<<endl;
        if(mini<k){
            return -1;
        }
        if(k<mini){
            return f.size();
        }
        return f.size()-1;
    }
};

File: Leetcode/3375. Minimum Operations to Make Array Values Equal to K.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        freq=Counter(nums)
        mf=0
        mini=float('inf')
        for i in freq:
            mf=max(mf,freq[i])
            mini=min(i,mini)
        if mini<k:
            return -1
        if k not in freq:
            return len(freq)
        return len(freq)-1
        
